src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -4374,6 +4387,17 @@ private DynamicObject makeRope(DynamicObject string, Rope rope, int beg, int byt
             return ret;
         }
 
+        protected static boolean indexTriviallyOutOfBounds(DynamicObject string, int index, int length) {
+            assert RubyGuards.isRubyString(string);
+
+            return (length < 0) || (index > rope(string).characterLength());
+        }
+
+        protected static boolean noCharacterSearch(DynamicObject string) {
```

Comment:
I would invert and name it `needsCharacterSearch` to avoid `!noCharacterSearch(string)`.

Reply:
Okay. I debated this. Keeping it negative made the boolean expression easier to read, IMO. But I'll change it.